### DISCUSSION TITLE
Cross-references between files

### DISCUSSION THREAD

**Discussion by ekatko1** [quarto-dev/quarto-cli#8955] [Q&A]
### Description

I think it would be great to be able to have cross-references between files outside of Quarto books. 
For example, when writing a manuscript, authors typically have supplemental materials - text, figures and tables that need to be cited within the manuscript, but shared in one or more separate files. 
As it stands, I do not see a way that would make this possible, except perhaps by abusing the book template in a way that is not really how it is intended to be used (I have not tried this).

This is sort of already adressed in #5452 but as a more general restructuring process.

**Comment by mcanouil** [comment]
Put your supplementary in the document and split afterwards.

A book is a single unit, even if it uses multiple files.
Same goes for websites.

I am pretty sure, there is no tools able to do cross files reference.

**Reply by ekatko1** [reply]
You are right. Putting everything in one file is probably the simplest approach. Do you know if there is a tool to split up the files automatically? My output is word documents.

**Comment by andrewpbray** [comment]
Hi @ekatko1 , what are you imagining in terms of a cross-reference here? What is the output format you'd be using? Have you played with the Manuscript project type?

**Reply by ekatko1** [reply]
I would imagine that cross-reference tags display the correct text e.g. "Supplemental Figure 2" or "Table S1", not necessarily provide links that open other files (though that would be pretty cool!) Then I could re-use the same tags in a different but linked document and make sure the tags produce the same labels / numbering. Pretty much just like things are in any publication. Nearly all have at least one supplemental Figure or table nowadays. 

I am using docx format, but this could be cool in more of a website context as well and perhaps even PDF. 

I have not yet tried the Manuscript project type but will no doubt do that sometime soon!

**Reply by mcanouil** [reply]
I am pretty convinced it's not possible.
For instance LaTeX/PDF, is doing the counting on compilation, there is no direct way for one LaTeX document to have access to the floats counts of another document.
Same goes for Microsoft Word, you cannot make a cross-reference to something in another document, you just write plain text.

The only way would be to completely take over the cross-reference system to turn actual cross-références into plain text when across multiple independent files and also keep some kind of index of the cross-referenceable objects.

It seems way easier and faster for you to make a single document Word/PDF that you split afterward when needed.

**Reply by ekatko1** [reply]
I think that spliting manually is a reasonable enough approach. However, it would be even better if the splitting could be automated. It seems to me like that would be something that may not be super complex to code, but feel free to correct me if I am wrong.

Another drawback of this method is that it takes longer to render a single large document than it does for multiple smaller docs, but not a dealbraker in most cases.

**Reply by mjsmith037** [reply]
Just for full transparency, it is possible to do this in straight LaTeX (i.e. not through quarto), using the xr package (which utilizes the .aux files from each file's respective compilation to get float numbers). I have been unable to get this workflow to work with two quarto documents, however, even when saving the aux and tex files

P.S. A use-case in which this is helpful is in submitting research articles to journals that require the SI to be submitted as a separate document than the main text, often with different formatting requirements

**Comment by iagopinal** [comment]
Hi, You can take a look at [figure-table-formatter](https://github.com/iagopinal/figure-table-formatter). You can cross-reference the figures and tables in the order that they are mentioned in the text, and sort them accordingly. It automatically outputs a file with the sorted tags of the figures and tables. You can use this file to automatically sort the figures in other files using the same filter (handy for supplements).

It would be nice to have the same functionality in the native Quarto, but at least this is a viable alternative now.

**Reply by cscheid** [reply]
@iagopinal Thanks for writing this! In the future, can you consider creating a new post at Announcements or Show and Tell instead of responding to many discussion threads? That cuts down on the volume for everyone. Thanks!

**Reply by iagopinal** [reply]
I'll do that the next time! Thanks for the great work that you do with Quarto!

**Comment by srgillott** [comment]
I just want to make sure I understand something before making a feature request. I am very new to LaTeX and Quarto so apologies in advance. 

@mcanouil says its impossible to cross reference different documents but if book projects can reference different .qmd files, then there must be a way to make a non book project project work the same? In addition, @mjsmith037 mentioned its possible. Is there some confusion on my part here about already rendered documents/language and the original question? 

I am using a journals manuscript template so cannot use the book project type and a requirement is to have a separate Supplementary Information (SI) document. My use case is fine to have them rendered at the same time (like a single unit book project) but as already mentioned having control of them to render separately to save compute time would be useful during the development of the work instead of a single document and separating them manually after. Much of my motivation to learn quarto is from the reproducible science perspective so all that is needed to do at the end is click run and everything is able to be recreated without further input. 

If there is the ability for allowing other non-book projects to cross reference different documents for tasks such as SI in paper writing then please let me know. If its a possible feature then i will make an appropriate request.

Thanks

**Reply by mcanouil** [reply]
@ekatko1 Please open a new thread referencing this one as this one is quite old, and when you are asking a question. This helps keep things tidy for future readers.
In your new thread, please explain what you are trying to do in what context exactly.
It would be useful to have a small reproducible example of what is not working exactly, such as a Git repository with `quarto create project` and relevant changes.

Manuscript are not books which are not website projects.
These projects have different designs, especially the Manuscript type. Manuscript is one single "standalone" main document that can have companions notebooks from which it can extracts components.
That's not supplementary files.

Also before opening anything be sure to look for the various thread about manuscript, etc. such as about cross-referencing files:

- https://github.com/quarto-dev/quarto-cli/discussions/2900
